src/libsyncengine/jobs/network/upload_session/loguploadsession.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Infomaniak kDrive - Desktop
+ * Copyright (C) 2023-2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "loguploadsession.h"
+#include "libparms/db/parmsdb.h"
+#include "log/log.h"
+
+namespace KDC {
+
+LogUploadSession::LogUploadSession(const SyncPath &filepath, uint64_t nbParalleleThread /*= 1*/)
+    : AbstractUploadSession(filepath, filepath.filename(), nbParalleleThread) {
+    _uploadSessionType = UploadSessionType::LogUpload;
+}
+
+bool LogUploadSession::runJobInit() {
+    return true;
+}
+
+std::shared_ptr<UploadSessionStartJob> LogUploadSession::createStartJob() {
+    return std::make_shared<UploadSessionStartJob>(UploadSessionType::LogUpload, getFileName(), getFileSize(), getTotalChunks());
+}
+
+std::shared_ptr<UploadSessionChunkJob> LogUploadSession::createChunkJob(const std::string &chunckContent, uint64_t chunkNb,
+                                                                        std::streamsize actualChunkSize) {
+    return std::make_shared<UploadSessionChunkJob>(UploadSessionType::LogUpload, getFilePath(), getSessionToken(), chunckContent,
+                                                   chunkNb, actualChunkSize, jobId());
+}
+
+std::shared_ptr<UploadSessionFinishJob> LogUploadSession::createFinishJob() {
+    SyncTime modtimeIn =
+        std::chrono::time_point_cast<std::chrono::seconds>(std::chrono::system_clock::now()).time_since_epoch().count();
+
+    return std::make_shared<UploadSessionFinishJob>(UploadSessionType::LogUpload, getFilePath(), getSessionToken(),
+                                                    getTotalChunkHash(), getTotalChunks(), modtimeIn);
+}
+
+std::shared_ptr<UploadSessionCancelJob> LogUploadSession::createCancelJob() {
+    return std::make_shared<UploadSessionCancelJob>(UploadSessionType::LogUpload, getSessionToken());
+}
+
+bool LogUploadSession::prepareChunkJob(const std::shared_ptr<UploadSessionChunkJob> &chunkJob) {
+    return true;
+}
+
+bool LogUploadSession::handleStartJobResult(const std::shared_ptr<UploadSessionStartJob> &StartJob, std::string uploadToken) {
+    AppStateValue appStateValue = "";
+    if (bool found = false; !ParmsDb::instance()->selectAppState(AppStateKey::LogUploadToken, appStateValue, found) || !found) {
+        LOG_WARN(getLogger(), "Error in ParmsDb::selectAppState");
```

Comment:
Error but no return `false`? This method never return `false`. is it normal?

Reply:
It is an error, but we do not wish to abort the operation. All errors occurring here are related to the cancellation of a previously failing log upload attempt. We will try to close it cleanly, but if we fail, it's not a significant issue (it will be closed automatically after x hours).